brainglobe_atlasapi/atlas_generation/mesh_utils.py

```diff
@@ -245,6 +247,86 @@ def create_region_mesh(args):
                 )
 
 
+def construct_meshes_from_annotation(
+    download_path, volume, structures_list, root_id
+):
+    """
+    Retrieve or construct atlas region meshes for a given annotation volume.
+
+    If an atlas is packaged with mesh files, reuse those. Otherwise, construct
+    the meshes using the existing volume and structure tree. Returns a
+    dictionary mapping structure IDs to their corresponding .obj mesh files.
+
+    Parameters
+    ----------
+    download_path : Path
+        Path to the directory where new mesh files will be saved.
+    volume : np.ndarray
+        3D annotation volume.
+    structures_list : list
+        List of structure dictionaries containing id information.
+    root_id : int
+        Identifier for the root structure.
+
+    Returns
+    -------
+    dict
+        Dictionary of structure IDs and paths to their .obj mesh files.
+    """
+    meshes_dir_path = download_path / "meshes"
+    meshes_dir_path.mkdir(exist_ok=True)
+
+    tree = get_structures_tree(structures_list)
+    labels = np.unique(volume).astype(np.int32)
+
+    for key, node in tree.nodes.items():
+        node.data = Region(key in labels)
+
+    closing_n_iters = 2
+    decimate_fraction = 0
+    smooth = False
```

Comment:
I wonder whether these should be exposed as parameters, with default values, so we can re-use this function across more packaging scripts (as and when we need to, moving to version 2)

Reply:
absolutely

Reply:
I exposed these parameters. I havent tested it yet as im currently generating demba files (and will be for a while!). There is the option to expose all of the parameters of extract mesh from mask but I'm not to familiar with that function.

Reply:
it runs :)